lib/pos_consensus.go

```diff
@@ -393,15 +393,24 @@ func (cc *FastHotStuffConsensus) HandleLocalTimeoutEvent(event *consensus.FastHo
 		return errors.Errorf("FastHotStuffConsensus.HandleLocalTimeoutEvent: Error advancing view on timeout: %v", err)
 	}
 
+	// Extract the tip block hash from the timeout message
+	tipBlockHash := BlockHashFromConsensusInterface(event.TipBlockHash)
+
+	// Fetch the HighQC from the Blockchain struct
+	tipBlockNode, tipBlockExists := cc.blockchain.blockIndexByHash[*tipBlockHash]
+	if !tipBlockExists {
+		return errors.Errorf("FastHotStuffConsensus.HandleLocalTimeoutEvent: Error fetching tip block: %v", tipBlockHash)
+	}
+
 	// Construct the timeout message
 	timeoutMsg := NewMessage(MsgTypeValidatorTimeout).(*MsgDeSoValidatorTimeout)
 	timeoutMsg.MsgVersion = MsgValidatorTimeoutVersion0
 	timeoutMsg.TimedOutView = event.View
 	timeoutMsg.VotingPublicKey = cc.signer.GetPublicKey()
-	timeoutMsg.HighQC = QuorumCertificateFromConsensusInterface(event.QC)
+	timeoutMsg.HighQC = QuorumCertificateFromConsensusInterface(tipBlockNode.Header.GetQC())
```

Comment:
Rather than using the QC from the timeout event, we now fetch the QC from the Blockchain struct. This will allow us to set a custom QC if needed for a synthetic genesis block